File: app/src/main/java/site/linyuange/android/samples/main/MainActivity.java
package site.linyuange.android.samples.main;

import android.databinding.DataBindingUtil;

import site.linyuange.android.samples.R;
import site.linyuange.android.samples.base.BaseActivity;
import site.linyuange.android.samples.databinding.ActivityMainBinding;
import site.linyuange.android.samples.rv.RecyclerViewActivity;

public class MainActivity extends BaseActivity implements Navigation {

    private ActivityMainBinding mBinding;

    @Override
    protected void initView() {
        mBinding = DataBindingUtil.setContentView(this, getLayoutRes());
        mBinding.setNavigator(this);
    }

    @Override
    protected int getLayoutRes() {
        return R.layout.activity_main;
    }

    @Override
    public void navToRvPage() {
        RecyclerViewActivity.startInstance(this);
    }
}

File: app/src/main/java/site/linyuange/android/samples/MainActivity.kt
package site.linyuange.android.samples

import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*
import site.linyuange.android.samples.recoder.ScreenRecorderActivity
import site.linyuange.framework.BaseActivity

/**
 * Author: BaHuang
 * Date: 2019/5/24 12:59
 */
class MainActivity : BaseActivity() {
    override fun getLayoutRes(): Int {
        return R.layout.activity_main
    }

    override fun initData() {
        mBtnMediaRecorder.setOnClickListener { naviToMediaRecorderPage() }
    }

    private fun naviToMediaRecorderPage() {
        Toast.makeText(this, "naviToMediaRecorderPage", Toast.LENGTH_LONG).show();
        ScreenRecorderActivity.getStartIntent(this@MainActivity)
    }
}